### PR TITLE
Replace Element.prototype.documentOffSetTop with jQuery's offset #3877

### LINKED ISSUE #3877: Replace Element.prototype.documentOffSetTop with jQuery's offset
Rule of the thumb: Do not modify objects you do not own, if there is a good reason for it, it should be done in a common file like common.js and be documented.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #3877 
@bamboo3250 should be easy to review :D

### CODE REVIEW

**Comment by bamboo3250** [comment]
good to merge :smile:

File: src/main/webapp/js/common.js [hunk 1]
```diff
@@ -441,11 +441,11 @@ function setStatusMessage(message, error) {
         $(DIV_STATUS_MESSAGE).attr('class', 'alert alert-warning');
     }
 
-    Element.prototype.documentOffsetTop = function() {
-        return this.offsetTop + (this.offsetParent ? this.offsetParent.documentOffsetTop() : 0);
+    function getDocumentOffsetTop(element) {
```
Comment (damithc) [inline src/main/webapp/js/common.js]: The function name is not completely self explanatory. Rename or add a comment?

**Comment by yj-soh** [comment]
On a side note, is there a reason why it's even creating a function when it can simply calculate and store the value in a variable there?

**Comment by wkurniawan07** [comment]
@yj-soh I'm guessing because there's recursion involved.

File: src/main/webapp/js/common.js (hunk 1 above)
Comment (wkurniawan07) [inline src/main/webapp/js/common.js]: Actually, this should be `getElementOffsetTopRelativeToDocument`.
Offset is an HTML lingo for the distance of the current element relative to another (usually its parent element). Should I add a comment to explain that?

Comment (damithc) [inline src/main/webapp/js/common.js]: Yes, rename as suggested (preferred) or add a comment.

**Comment by yj-soh** [comment]
Actually, isn't this just a re-implementation of jQuery's `offset().top`?
Can you try changing that line to the following and see if it does the same thing?
`var positionToScrollTo = $(DIV_STATUS_MESSAGE).offset().top - (window.innerHeight / 2);`

**Comment by wkurniawan07** [comment]
It works! Amazing. The `window.innerHeight / 2` is to make the status message centered, so we still need it.

**Comment by wkurniawan07** [comment]
In light of how the issue is fixed, I should change the issue name, just like #3878.

**Pull request merged**